### ISSUE TITLE
Windows: std.fs.Dir.deleteDir() will silently fail on non-empty directories

### ISSUE THREAD

**Issue by squeek502** [ziglang/zig#5537]
When `./dir` is non-empty, the following code will succeed but the directory will not be deleted. I expect `error.DirNotEmpty` to be returned in this case.

```zig
const std = @import("std");

pub fn main() !void {
    return std.fs.cwd().deleteDir("dir");
}
```

Both NtCreateFile and NtClose seem to be returning `SUCCESS` here:

https://github.com/ziglang/zig/blob/fd067fbe8b14c445ed764da15fff5163156126df/lib/std/os.zig#L1768-L1791

**Comment by squeek502** [comment]
From https://go.microsoft.com/fwlink/?LinkId=140636 section 4.3.1 (page 33):

> Note that if the DELETE_ON_CLOSE flag is specified on a directory with child files or directories, the create operation will succeed, but the delete on close flag will be silently ignored when processing the cleanup IRP and the directory will not be deleted.

**Comment by kubkon** [comment]
Should be fixed with #6397 so closing. If the issue persists, please feel free to reopen!

**Comment by squeek502** [comment]
Should be re-opened as #6397 was reverted in #6462

(for some reason I can't re-open it even though I started this issue?)
